src/Security/MemberAuthenticator/SessionAuthenticationHandler.php

```diff
@@ -72,13 +75,20 @@ public function logIn(Member $member, $persistent = false, HTTPRequest $request
     {
         static::regenerateSessionId();
         $request = $request ?: Controller::curr()->getRequest();
-        $request->getSession()->set($this->getSessionVariable(), $member->ID);
+        $session = $request->getSession();
+        $session->set($this->getSessionVariable(), $member->ID);
 
         // This lets apache rules detect whether the user has logged in
         // @todo make this a setting on the authentication handler
         if (Member::config()->get('login_marker_cookie')) {
             Cookie::set(Member::config()->get('login_marker_cookie'), 1, 0);
         }
+
+        // Activate sudo mode on login so the user doesn't have to reauthenticate for sudo
+        // actions until the sudo mode timeout expires
+        /** @var SudoModeServiceInterface $service */
+        $service = Injector::inst()->get(SudoModeServiceInterface::class);
+        $service->activate($session);
```

Comment:
Note: I opted to move this here instead of `SilverStripe\Security\MemberAuthenticator\LoginHandler` where the original extension was applied for the following reasons:
- Makes it WAY easier to test
- Ensures it is always added on login, not only if using the above handler which is only activated when logging in via a form
- Sudo Mode status is stored in the session, so if this handler (`SessionAuthenticationHandler`) isn't available it probably means the devs are doing something screwy with sessions so sudo mode probably won't work anyway.